src/main/java/in/ac/bits/protocolanalyzer/mvc/controller/TestController.java

```diff
@@ -38,24 +40,22 @@
 	@Qualifier("Experiment")
 	private Experiment exp1;
 
-//	@RequestMapping(method = RequestMethod.GET)
-	public void runSequentialExp()throws Exception {
-		String pcapPath = "/home/vikram/Darshini/Test_pcap_files/sample_capture_test1.pcap";
-		String protocolGraphStr = "graph start {\n\tethernet;\n}\ngraph ethernet {\n\tswitch(ethertype) {\n\t\tcase 0800:			 ipv4;\n\t}\n}\ngraph ipv4 {\n\tswitch(protocol) {\n\t\tcase 06: tcp;\n\t}\n}\ngraph tcp {\n}\ngraph end {\n}";
-
+	@RequestMapping(value = "/experiment", method = RequestMethod.GET)
+	public @ResponseBody String runSequentialExp(@RequestParam("protocolGraphPath") String protocolGraphPath, @RequestParam("pcapPath") String pcapPath)throws Exception {
+	    String protocolGraphStr = new String(Files.readAllBytes(Paths.get(protocolGraphPath)));
```

Comment:
You are using File I/O here. This section needs proper exception handling for file not existing, file permission errors, null string scenarios. For all these scenarios, you can simply return count packet zero from this function.

Even for a zero-sized file / file permissions error / file not existing scenarios of pcapPath, we can return zero packet count from this function. In the case of pcapPath, we should not read the file; we should only test it for file permissions error.